_posts/2024/2024-02-06-our-process.md

```diff
@@ -0,0 +1,453 @@
+Back in 2022 I wrote that there are two ways of looking at roles in a small-to-medium tech company:
+what people do, and [what risks they address][mitigation].
+Coincidentally,
+I had just started working with someone who has turned out to be
+[the best product manager I've ever met][dottie].
+In the two years since she joined the company
+she has overseen lots of improvements in our software engineering processes
+(partly, I suspect, to make it easier for her to do her job properly).
+She recently summarized the state of play to orient a new hire;
+what follows is a summary of her summary with a few explanatory notes
+that I hope will be useful to other software developers
+working in research-intensive organizations.
+
+I believe the most important part of what follows
+is how she has organized her thinking.
+Each process exists *to answer key questions*,
+has its own *frequency*,
+is *someone's responsibility*,
+and generates *specific products*.
+If you have a description like this of how your team operates,
+please [get in touch](mailto:{{site.author.email}}):
+I think we could learn a lot from comparisons.
+
+## Company Goals and Priority Projects
+
+This activity happens up in the stratosphere.
+As an engineering manager I may occasionally be called on for input,
+but for the most part this is the ground for everything else described below.
+
+-   Summary: Prioritized company objectives and initiatives for the year
+-   Key Questions:
+    -   What should we invest in/focus on?
+-   Frequency: 6-24 months
+-   Responsible: executive team 
+-   Contributors: company leadership and technical experts at all levels
+-   Products:
+    -   Monthly business review
+    -   Annual all-hands meeting
+
+## Engineering Team
+
+Our team includes about a dozen people
+and is responsible for half that number of applications,
+ranging from complex C++ libraries wrapped up in Python
+to three-tier web applications for managing experimental data
+and command-line tools to support routine tasks.
+The activities below are done by the whole team.
+
+### Engineering OKRs
+
+The VP of Engineering defines objectives and key results at the start of each quarter
+based on discussions with senior management,
+the leaders of other teams,
+and members of our team.
+Goals are prioritized as P0 (not doing this will block something critical to the company),
+P1 (we need this, but we can scale it back or it can slip if a P0 overruns),
+and P2 (it would be nice to have).
+
+-   Summary: Engineering-specific objectives and projects based on:
+    1.  company goals
+    2.  users' needs
+    3.  technical opportunities
+-   Key Questions:
+    -   What should each engineering team focus on and why?
+-   Frequency: quarterly
+-   Responsible: VP Engineering
+-   Contributors: entire engineering team
+-   Products:
+    -   Prioritized list of objectives and key results
+
+> Note: we used to define more of these each quarter than we do now.
```

Comment:
Just a note: This didn't read very clear to me. I believe you're referring to how we went down from a list of 30+ tasks to a handful of OKRs? Or just generally that we keep it to a reasonable number of OKRs?

Reply:
updated